ImGuiFileDialog.cpp

```diff
@@ -1,72 +1,100 @@
-#include "ImGuiFileDialog.h"
+// From https://github.com/aiekick/ImGuiFileDialog
+//
+// Modified by AlastairGrowcott@yahoo.com to work on Posix systems.
+
+
+#include <sys/types.h>
+#include <dirent.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+
+#include "imgui_file_dlg.h"
 #include "imgui.h"
-#include "dirent/dirent.h"
 
 #define IMGUI_DEFINE_MATH_OPERATORS
 #include "imgui_internal.h"
 
-#include <algorithm>
 
-inline bool ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
```

Comment:
The return value is not used so I removed it to simplify the code. This is not a significant change, but I wasn't expecting to offer a pull request and my usual code tidying habits kicked in.